src/index.js

```diff
@@ -4,556 +4,608 @@ const codec = require('./codec')
+
+      i += (size + n)
+      codes.push(code)
+    }
 
-/**
- * Returns the names of the protocols in left-to-right order.
- * [See list of protocols](https://github.com/multiformats/multiaddr/blob/master/protocols.csv)
- *
- * @returns {Array.<string>} protocol names
- * @example
- * Multiaddr('/ip4/127.0.0.1/tcp/4001').protoNames()
- * // [ 'ip4', 'tcp' ]
- */
-Multiaddr.prototype.protoNames = function protoNames () {
-  return this.protos().map(proto => proto.name)
-}
+    return codes
+  }
 
-/**
- * Returns a tuple of parts
- *
- * @returns {[number, Uint8Array][]} tuples
- * @example
- * Multiaddr("/ip4/127.0.0.1/tcp/4001").tuples()
- * // [ [ 4, <Buffer 7f 00 00 01> ], [ 6, <Buffer 0f a1> ] ]
- */
-Multiaddr.prototype.tuples = function tuples () {
-  return codec.bytesToTuples(this.bytes)
-}
+  /**
+   * Returns the names of the protocols in left-to-right order.
+   * [See list of protocols](https://github.com/multiformats/multiaddr/blob/master/protocols.csv)
+   *
+   * @returns {Array.<string>} protocol names
+   * @example
+   * ```js
+   * Multiaddr('/ip4/127.0.0.1/tcp/4001').protoNames()
+   * // [ 'ip4', 'tcp' ]
+   * ```
+   */
+  protoNames () {
+    return this.protos().map(proto => proto.name)
+  }
 
-/**
- * Returns a tuple of string/number parts
- * - tuples[][0] = code of protocol
- * - tuples[][1] = contents of address
- *
- * @returns {[number, string|number][]} tuples
- * @example
- * Multiaddr("/ip4/127.0.0.1/tcp/4001").stringTuples()
- * // [ [ 4, '127.0.0.1' ], [ 6, 4001 ] ]
- */
-Multiaddr.prototype.stringTuples = function stringTuples () {
-  const t = codec.bytesToTuples(this.bytes)
-  return codec.tuplesToStringTuples(t)
-}
+  /**
+   * Returns a tuple of parts
+   *
+   * @returns {[number, (Uint8Array | undefined)?][]} tuples
+   * @example
+   * ```js
+   * Multiaddr("/ip4/127.0.0.1/tcp/4001").tuples()
+   * // [ [ 4, <Buffer 7f 00 00 01> ], [ 6, <Buffer 0f a1> ] ]
+   * ```
+   */
+  tuples () {
+    return codec.bytesToTuples(this.bytes)
+  }
 
-/**
- * Encapsulates a Multiaddr in another Multiaddr
- *
- * @param {Multiaddr} addr - Multiaddr to add into this Multiaddr
- * @returns {Multiaddr}
- * @example
- * const mh1 = Multiaddr('/ip4/8.8.8.8/tcp/1080')
- * // <Multiaddr 0408080808060438 - /ip4/8.8.8.8/tcp/1080>
- *
- * const mh2 = Multiaddr('/ip4/127.0.0.1/tcp/4001')
- * // <Multiaddr 047f000001060fa1 - /ip4/127.0.0.1/tcp/4001>
- *
- * const mh3 = mh1.encapsulate(mh2)
- * // <Multiaddr 0408080808060438047f000001060fa1 - /ip4/8.8.8.8/tcp/1080/ip4/127.0.0.1/tcp/4001>
- *
- * mh3.toString()
- * // '/ip4/8.8.8.8/tcp/1080/ip4/127.0.0.1/tcp/4001'
- */
-Multiaddr.prototype.encapsulate = function encapsulate (addr) {
-  addr = Multiaddr(addr)
-  return Multiaddr(this.toString() + addr.toString())
-}
+  /**
+   * Returns a tuple of string/number parts
+   * - tuples[][0] = code of protocol
+   * - tuples[][1] = contents of address
+   *
+   * @returns {[number, string?][]} tuples
```

Comment:
```suggestion
   * @returns {[number, string|number?][]} tuples
```

Reply:
this is one of the "breaking changes",now at least the value is always a string.

`tuplesToStringTuples` was documented as `[string,string][]` but the tests expect `[number, string|number][]` so i made it at least `[number, string][]` because index 2 comes from `convert.toString` (port was the only thing not convert to string).

Reply:
That sounds good to me. I noticed this because of the example below. We can do the breaking change, just need to update the example below to have the port as a string